### ISSUE TITLE
ONNX export failure: Exporting the operator hardsigmoid to ONNX opset version 13 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.

### ISSUE THREAD

**Issue by ZS-YANG** [meituan/YOLOv6#880]
### Before Asking

- [X] I have read the [README](https://github.com/meituan/YOLOv6/blob/main/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [X] I want to train my custom dataset, and I have read the [tutorials for training your custom data](https://github.com/meituan/YOLOv6/blob/main/docs/Train_custom_data.md) carefully and organize my dataset correctly; (FYI: We recommand you to apply the config files of xx_finetune.py.) 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。（FYI: 我们推荐使用xx_finetune.py等配置文件训练自定义数据集。）

- [X] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [X] I have searched the YOLOv6 [issues](https://github.com/meituan/YOLOv6/issues) and found no similar questions.

### Question

1. 当我将yolov6lite_s.pt模型转换为onnx模型时，报如下错误

Starting to export ONNX...
/home/yangzhanshan/disk/anaconda3/envs/yolov6/lib/python3.8/site-packages/torch/tensor.py:587: RuntimeWarning: Iterating over a tensor might cause the trace to be incorrect. Passing a tensor of different shape won't change the number of iterations executed (and might lead to errors or silently give incorrect results).
  warnings.warn('Iterating over a tensor might cause the trace to be incorrect. '
ONNX export failure: Exporting the operator hardsigmoid to ONNX opset version 13 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.

2. 报这个错的原因为onnx不支持nn.Hardsigmoid()激活函数的转换。

3. 去网上搜索解决办法，有人说可以通过 [将代码中的 hardsigmoid 函数替换为 hardtanh函数解决](https://codeleading.com/article/67586407296/)
4. 请问这个解决办法会影响精度嘛，以及官方会修复这个bug嘛

**Comment by gyftsy** [comment]
Hello, thank you for using yolov6-lite. This problem will indeed be encountered in the conversion of onnx. At present, this problem is mainly caused by the fact that this operator is not supported by onnx. We also solved this problem by replacing the hardsigmoid function in the code with the hardtanh function. We don't have a better solution at the moment. The model accuracy and speed measurement results shown so far are based on the measurements obtained by this conversion, and should not affect the accuracy of the model. If you have a better solution. Welcome to sync to us, thanks!

**Comment by ZS-YANG** [comment]
> Hello, thank you for using yolov6-lite. This problem will indeed be encountered in the conversion of onnx. At present, this problem is mainly caused by the fact that this operator is not supported by onnx. We also solved this problem by replacing the hardsigmoid function in the code with the hardtanh function. We don't have a better solution at the moment. The model accuracy and speed measurement results shown so far are based on the measurements obtained by this conversion, and should not affect the accuracy of the model. If you have a better solution. Welcome to sync to us, thanks!

thanks 👏

**Comment by pengshuaibang** [comment]
![f7245611e68c28951771682ae9f595e0](https://github.com/meituan/YOLOv6/assets/152959842/8bf1565d-e6d3-48ab-a7a8-b96de7ae04a3)
![fdda2d38d5f5d74d63bf8c1038adf07b](https://github.com/meituan/YOLOv6/assets/152959842/d65216c2-6320-4b9b-ab0b-9bbc959b0364)
只需要找到activation.py脚本，然后找到Hardsigmoid类，修改以下的代码：
    def forward(self, input: Tensor) -> Tensor:
        return F.hardtanh(input + 3, 0., 6., self.inplace)# for torchscript, CoreML and ONNX
        # return F.hardsigmoid(input, self.inplace)原代码# for torchscript and CoreML

**Comment by ZS-YANG** [comment]
> ![f7245611e68c28951771682ae9f595e0](https://private-user-images.githubusercontent.com/152959842/287963566-8bf1565d-e6d3-48ab-a7a8-b96de7ae04a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzODE3OTYsIm5iZiI6MTcwMjM4MTQ5NiwicGF0aCI6Ii8xNTI5NTk4NDIvMjg3OTYzNTY2LThiZjE1NjVkLWU2ZDMtNDhhYi1hN2E4LWI5NmRlN2FlMDRhMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxMlQxMTQ0NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNmMxNTQ0OWFjNTNlNGFkZTgzM2QxMTczMWMzM2UyZDNmZjc3MDI1YzI3NzE0ZWZmN2JiMGQxZjVmNTljNjNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.afWTeyaS8J3Je3Ba3trcaqA9rvDOsx9_pIlUbVlqryw) ![fdda2d38d5f5d74d63bf8c1038adf07b](https://private-user-images.githubusercontent.com/152959842/287963576-d65216c2-6320-4b9b-ab0b-9bbc959b0364.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzODE3OTYsIm5iZiI6MTcwMjM4MTQ5NiwicGF0aCI6Ii8xNTI5NTk4NDIvMjg3OTYzNTc2LWQ2NTIxNmMyLTYzMjAtNGI5Yi1hYjBiLTliYmM5NTliMDM2NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxMlQxMTQ0NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mOTk2MWUzMTJmM2M1MzA1MTBhYmNmMWQ1MDE1ZTBhMTY5NTI1ZGU5ZTE4NjdiNjA3NGExMjU3MGI3MWIxMjE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EqBS5uGgoo7kdA1QUDaIZ0C-deqTVwDegCplEqUTZ3w) 只需要找到activation.py脚本，然后找到Hardsigmoid类，修改以下的代码： def forward(self, input: Tensor) -> Tensor: return F.hardtanh(input + 3, 0., 6., self.inplace)# for torchscript, CoreML and ONNX # return F.hardsigmoid(input, self.inplace)原代码# for torchscript and CoreML @ZS-YANG @gyftsy

thx.
